django/contrib/admin/checks.py

```diff
@@ -806,10 +806,10 @@ def _check_list_editable_item(self, obj, model, field_name, label):
                         id='admin.E123',
                     )
                 ]
-            # Check that list_display_links is set, and that the first values of list_editable and list_display are
-            # not the same. See ticket #22792 for the use case relating to this.
-            elif (obj.list_display[0] in obj.list_editable and obj.list_display[0] != obj.list_editable[0] and
-                  obj.list_display_links is not None):
+            # If list_display[0] is is list_editable, check that
+            # list_display_links is set. See ticket #22792 and #26229 for the use
+            # cases relating to this.
+            elif (obj.list_display[0] == field_name and not obj.list_display_links and obj.list_display_links is not None):
```

Comment:
We call this method `_check_list_editable_item` for every field name in list_editable. Comparing `obj.list_display[0] == field_name` instead of `obj.list_display[0] in obj.list_editable` means that we can include the correct label in the error message.

The extra and not `obj.list_display_links` is required now that we don't have the `obj.list_display[0] != obj.list_editable[0]` check. If obj.list_display_links is a non-empty iterable, then we don't have to worry about the value of `list_display[0]`. There is a separate check admin.E123 to make sure that a field is not in `list_display_links` and `list_editable`.